### PR TITLE
Allows Grilling in Space (SBO28)

### LINKED ISSUE #21087: Cant place windows on an adjacent grille if you're on a space tile.
#### Description of issue
If the tile you're standing on isnt a valid location to create a window (space for example) you cannot put a window on a grille that has proper floor support underneath it that is next to you.

[bugfix]

#### Difference between expected and actual behavior
Window attaching itself to the grille.
Error message about floor support.

#### Steps to reproduce
Attempt to replace a broken window from space by clicking on a grille.

#### Length of time in which bug has been known to occur

since https://github.com/vgstation-coders/vgstation13/pull/20611 i'd imagine.

### PR DESCRIPTION
**Pull request by Kurfursten**
Improved the build system, you may now specify a build location.

fixes #21087

🆑 
* bugfix: You can properly build a window on a valid tile (e.g.: a grille) even if you are not on a valid tile (e.g.: space).

### CODE REVIEW

**Comment by Kurfursten** [comment]
@Exxion

File: code/game/objects/items/stacks/stack_recipes.dm [hunk 1]
```diff
@@ -37,14 +37,16 @@
 /datum/stack_recipe/proc/finish_building(var/mob/usr, var/obj/item/stack/S, var/R) //This will be called after the recipe is done building, useful for doing something to the result if you want.
 	return R
 
-/datum/stack_recipe/proc/build(var/mob/usr, var/obj/item/stack/S, var/multiplier = 1)
+/datum/stack_recipe/proc/build(var/mob/usr, var/obj/item/stack/S, var/multiplier = 1, var/turf/construct_loc)
```
Comment (Exxion) [inline code/game/objects/items/stacks/stack_recipes.dm:40]: >`usr` as a param

God what the fuck

Comment (Kurfursten) [reply code/game/objects/items/stacks/stack_recipes.dm:40]: I noticed it too but it's used throughout. I just assumed bauds knew what he was doing 3 years ago and chose that for a very good reason

Comment (Exxion) [reply code/game/objects/items/stacks/stack_recipes.dm:40]: @I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE

Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [reply code/game/objects/items/stacks/stack_recipes.dm]: I have no idea what this is. This is used by Topic() though so it always uses usr at the moment, if the parameter name causes problems just like rename it to user aka totally_not_usr like 99% of verbs and Topic() does

Comment (Exxion) [reply code/game/objects/items/stacks/stack_recipes.dm:40]: `user` is the norm
And using this var name precludes the use of the actual `usr` var, and might potentially maybe even override it in procs it calls? Don't know about that one

**Pull request merged**